hoot-core/src/main/cpp/hoot/core/algorithms/WayMatchStringMerger.h

```diff
@@ -55,60 +55,57 @@ class WayMatchStringMerger
   {
   public:
 
-    WayPtr newWay1;
-    WaySubline subline2;
-    ConstWayPtr way2;
-
     WayLocation getEnd1() const { return _end; }
-    WayLocation getEnd2() const { return subline2.getEnd(); }
+    WayLocation getEnd2() const { return _subline2.getEnd(); }
+    WayLocation getEnd(WayNumber way) { return (way == WayNumber::Way1) ? getEnd1() : getEnd2(); }
```

Comment:
Helper function to convert the generic call of `getEnd` to `getEnd1` or `getEnd2` based on the `WayNumber` parameter.